crates/hir_def/src/generics.rs

```diff
@@ -340,27 +346,29 @@ impl GenericParams {
         })
     }
 }
-
-impl HasChildSource for GenericDefId {
-    type ChildId = LocalTypeParamId;
-    type Value = Either<ast::Trait, ast::TypeParam>;
-    fn child_source(&self, db: &dyn DefDatabase) -> InFile<SourceMap> {
-        let (_, sm) = GenericParams::new(db, *self);
-        sm
+impl GenericDefId {
+    // FIXME: this shouldn't exist?
+    pub fn child_source(&self, db: &dyn DefDatabase) -> InFile<SourceMaps> {
+        GenericParams::new(db, *self).1
```

Comment:
Unfortunately `HasChildSource` is actually being used by `TypeParam`(and for now also `LifetimeParam`) in their `HasSource` impls(can be seen in the diffs here), but as explained in this [zulip channel](https://rust-lang.zulipchat.com/#narrow/stream/185405-t-compiler.2Fwg-rls-2.2E0/topic/.236787.20lifetime.20reference.20search.20support) `HasChildSource` can't be used anymore here. Are there better approaches to solve this problem here?